experiments/pam/pam_cortx.c

```diff
@@ -72,11 +72,12 @@ static int login(pam_handle_t *pamh, const char *pUsername, const char *pPasswor
 	json_object_object_add(jobj, "password", json_object_new_string(pPassword));
 	/* set curl options */
 	curl_easy_setopt(pCurl, CURLOPT_URL, pUrl);
+	curl_easy_setopt(pCurl, CURLOPT_FAILONERROR, TRUE);
```

Comment:
We are not checking for errors in any of the `curl_easy_setopt()` calls (to say the least).  Any function that might return an error code must be checked.

Reply:
@tb-s Thanks for pointing out, as a part of this POC, we were trying to explore how to use REST api in PAM authentication. This is just a sample code for that. we are going to rewrite this entire PAM module in upcoming sprint.